test/e2e/ownerreference_test.go

```diff
@@ -0,0 +1,282 @@
+	vSphereClusterIdentityOwner = metav1.OwnerReference{Kind: "VSphereClusterIdentity", APIVersion: infrav1.GroupVersion.String()}
+
+	// CAPI owners.
+	clusterClassOwner       = metav1.OwnerReference{Kind: "ClusterClass", APIVersion: clusterv1.GroupVersion.String()}
+	clusterOwner            = metav1.OwnerReference{Kind: "Cluster", APIVersion: clusterv1.GroupVersion.String()}
+	clusterController       = metav1.OwnerReference{Kind: "Cluster", APIVersion: clusterv1.GroupVersion.String(), Controller: pointer.Bool(true)}
+	machineController       = metav1.OwnerReference{Kind: "Machine", APIVersion: clusterv1.GroupVersion.String(), Controller: pointer.Bool(true)}
+	clusterResourceSetOwner = metav1.OwnerReference{Kind: "ClusterResourceSet", APIVersion: addonsv1.GroupVersion.String()}
+
+	// KCP owner.
+	kubeadmControlPlaneController = metav1.OwnerReference{Kind: "KubeadmControlPlane", APIVersion: controlplanev1.GroupVersion.String(), Controller: pointer.Bool(true)}
+
+	// CAPBK owner.
+	kubeadmConfigController = metav1.OwnerReference{Kind: "KubeadmConfig", APIVersion: bootstrapv1.GroupVersion.String(), Controller: pointer.Bool(true)}
+)
+
+// The following names are hardcoded in templates to make cleanup easier.
+var (
+	clusterIdentityName            = "identity"
+	clusterIdentitySecretNamespace = "capv-system"
+	deploymentZoneName             = "zone1"
+)
+
+// cleanupVSphereObjects deletes the Secret, VSphereClusterIdentity, and VSphereDeploymentZone created for this test.
+// The VSphereFailureDomain, and the Secret for the VSphereClusterIdentity should be deleted as a result of the above.
+func cleanupVSphereObjects(ctx context.Context, bootstrapClusterProxy framework.ClusterProxy) bool {
+	Eventually(func() error {
+		if err := bootstrapClusterProxy.GetClient().Delete(ctx,
+			&infrav1.VSphereClusterIdentity{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterIdentityName,
+				},
+			}); err != nil {
+			return err
+		}
+		return bootstrapClusterProxy.GetClient().Delete(ctx,
+			&infrav1.VSphereDeploymentZone{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: deploymentZoneName,
+				},
+			})
+	}).Should(Succeed())
+	return true
+}
+
+func createVsphereIdentitySecret(ctx context.Context, bootstrapClusterProxy framework.ClusterProxy) bool {
+	credentialsSecret := &corev1.Secret{}
+	Expect(bootstrapClusterProxy.GetClient().Get(ctx, ctrlclient.ObjectKey{Namespace: "capv-system", Name: "capv-manager-bootstrap-credentials"}, credentialsSecret)).To(Succeed())
+	data := credentialsSecret.Data["credentials.yaml"]
+	credentials := map[string]string{}
+	Expect(yaml.Unmarshal(data, &credentials)).To(Succeed())
+	Expect(bootstrapClusterProxy.GetClient().Create(ctx,
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: clusterIdentitySecretNamespace,
+				Name:      clusterIdentityName,
+			},
+			Data: map[string][]byte{
+				"password": []byte(credentials["password"]),
+				"username": []byte(credentials["username"]),
```

Comment:
Up to you. Something like this should also work: `e2eConfig.GetVariable("VSPHERE_USERNAME")`